index.d.ts

```diff
@@ -0,0 +1,92 @@
+export class Client {
+  public config: Configuration;
+
+  constructor(apiKeyOrConfig: string | Configuration);
+
+  public notify(
+    error: Error,
+    beforeSendCallback?: BeforeSend,
+    blocking?: boolean,
+    postSendCallback?: (sent: boolean) => void,
+  ): void;
+
+  public setUser(id: string, name: string, email: string): void;
+
+  public clearUser(): void;
+
+  public leaveBreadcrumb(name: string, metadata?: IMetadata | string): void;
+}
+
+export class Configuration {
+  public version: string;
+  public apiKey?: string;
+  public delivery: IStandardDelivery;
+  public beforeSendCallbacks: BeforeSend[];
+  public notifyReleaseStages?: string[];
+  public releaseStage?: string;
+  public appVersion?: string;
+  public codeBundleId?: string;
+  public autoNotify: boolean;
+  public handlePromiseRejections: boolean;
+
+  constructor(apiKey?: string);
+
+  public shouldNotify(): boolean;
+
+  public registerBeforeSendCallback(callback: BeforeSend): void;
+
+  public unregisterBeforeSendCallback(
+    callback: BeforeSend,
+  ): void;
+
+  public clearBeforeSendCallbacks(): void;
+
+  public toJSON(): any;
+}
+
+type BeforeSend = (report: Report) => boolean | void;
+
+export class IStandardDelivery {
+  public endpoint: string;
+
+  constructor(endpoint: string);
+}
+
+export interface IMetadata {
+  type?:
+    | "error"
+    | "log"
+    | "navigation"
+    | "process"
+    | "request"
+    | "state"
+    | "user"
+    | "manual";
+  [key: string]: IMetadataValue | string | number | boolean | undefined;
+}
+
+export interface IMetadataValue {
+  [key: string]: string | number | boolean | undefined;
+}
+
+export class Report {
+  public apiKey: string;
+  public errorClass: string;
+  public errorMessage: string;
+  public context?: string;
+  public groupingHash?: string;
+  public metadata: IMetadata;
+  public severity: "warning" | "error" | "info";
+  public stacktrace: string;
+  public user: any;
+
+  constructor(apiKey: string, error: Error);
+
+  public addIMetadata(
```

Comment:
This function should not be renamed either, calling `addIMetdata` will fail at runtime. Also not usual practice to specify capital `I` for interfaces in functions that take them as parameters.